src/app-layer.c

```diff
@@ -706,6 +721,16 @@ void AppLayerSetupCounters(uint8_t ipproto)
                 snprintf(applayer_counters[ipproto_map][alproto].name,
                          strlen(str) + alproto_len + strlen(ipproto_suffix),
                          "%s%s%s", str, alproto_str, ipproto_suffix);
+                if (AppLayerParserProtocolIsTxAware(ipproto, alproto)) {
+                    applayer_counters[ipproto_map][alproto].tx_name =
+                        SCMalloc(strlen(tx_str) + alproto_len + strlen(ipproto_suffix));
```

Comment:
please use a 'size' variable here instead of "strlen(tx_str) + alproto_len + strlen(ipproto_suffix)" multiple times